core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -398,6 +398,7 @@ public void stop() {
         } finally {
             containerId = null;
             containerInfo = null;
+            portBindings.clear();
```

Comment:
if you clear the port bindings, then next start will obviously not have them. This does not sound like a fix to me.

We do set `containerId` and `containerInfo` to null here because they are obtained after the container is started, but `portBindings` is a part of the configuration.

Reply:
Thank you for your comment, now I see why this is not a good solution.